code/messaging/src/test/java/com/adobe/marketing/mobile/MessageTests.java

```diff
@@ -0,0 +1,360 @@
+*/
+
+package com.adobe.marketing.mobile;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import android.app.Activity;
+
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.services.ui.AEPMessage;
+import com.adobe.marketing.mobile.services.ui.AEPMessageSettings;
+import com.adobe.marketing.mobile.services.ui.FullscreenMessage;
+import com.adobe.marketing.mobile.services.ui.FullscreenMessageDelegate;
+import com.adobe.marketing.mobile.services.ui.MessageCreationException;
+import com.adobe.marketing.mobile.services.ui.MessageSettings;
+import com.adobe.marketing.mobile.services.ui.UIService;
+import com.adobe.marketing.mobile.services.ui.internal.MessagesMonitor;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Event.class, MobileCore.class, App.class, MessagingState.class, ServiceProvider.class})
+public class MessageTests {
+
+    private static final String html = "<html><head></head><body bgcolor=\"black\"><br /><br /><br /><br /><br /><br /><h1 align=\"center\" style=\"color: white;\">IN-APP MESSAGING POWERED BY <br />OFFER DECISIONING</h1><h1 align=\"center\"><a style=\"color: white;\" href=\"adbinapp://cancel\" >dismiss me</a></h1></body></html>";
+    private final Map<String, Object> consequence = new HashMap<>();
+
+    @Mock
+    Core mockCore;
+    @Mock
+    Activity mockActivity;
+    @Mock
+    MessagesMonitor mockMessagesMonitor;
+    @Mock
+    AndroidPlatformServices mockPlatformServices;
+    @Mock
+    UIService mockUIService;
+    @Mock
+    AEPMessage mockAEPMessage;
+    @Mock
+    Message mockMessage;
+    @Mock
+    AEPMessageSettings mockAEPMessageSettings;
+    @Mock
+    MessagingState mockMessagingState;
+    @Mock
+    MessagingInternal mockMessagingInternal;
+    @Mock
+    ServiceProvider mockServiceProvider;
+    @Captor
+    ArgumentCaptor<MessagingEdgeEventType> messagingEdgeEventTypeArgumentCaptor;
+    @Captor
+    ArgumentCaptor<String> interactionArgumentCaptor;
+    private Message message;
+    private AEPMessage aepMessage;
+    private EventHub eventHub;
+
+    @Before
+    public void setup() {
+        PowerMockito.mockStatic(MobileCore.class);
```

Comment:
nit: setup function has become really big and doing lot of things. Can we create functions and call from here.
like:
`mock()`
`setxdmmap()`